examples/ui_actions_explorer/public/context_menu_examples/panel_edit.tsx

```diff
@@ -0,0 +1,154 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import * as React from 'react';
+import { EuiButton, EuiContextMenu, EuiPopover } from '@elastic/eui';
+import useAsync from 'react-use/lib/useAsync';
+import { buildContextMenuForActions } from '../../../../src/plugins/ui_actions/public';
+
+export const PanelEdit: React.FC = () => {
+  const [open, setOpen] = React.useState(false);
+
+  const context = {};
+  const trigger: any = 'TEST_TRIGGER';
+
+  const panels = useAsync(() =>
+    buildContextMenuForActions({
+      actions: [
+        {
+          action: {
```

Comment:
In unit tests and in example samples there are A LOT of such inlined example actions. 

Since most of those required fields are duplicated, I'd suggest to at least have a helper `createSampleAction` function in each file, where it makes sense. If it is possible to reduce this test helper, would be even better. 

I'd put it some like this: 

```
const createSampleAction = ({type?, icon?,name? ,order? }, otherOverrides: Partial<ActionDefinition> = {}) =>  ({
             id: type,
             type: type as any,
             order: order,
             getDisplayName: () => name,
             getIconType: () => icon,
             isCompatible: async () => true,
             execute: async () => {},
            ...otherOverrides
           })
```

Adding and using something like this would help us maintain those tests and samples. For example:
*  if we add some new required field to action definition, we'd likely have to only update that function, instead of updating all the inclined occurrences. 
* actual test data would be focused on important stuff for the specific test, because things that not important would be omitted.

Reply:
Done.